src/tini.c

```diff
@@ -133,8 +133,19 @@ int spawn(const signal_configuration_t* const sigconf_ptr, char* const argv[], i
 		}
 
 		execvp(argv[0], argv);
-		PRINT_FATAL("Executing child process '%s' failed: '%s'", argv[0], strerror(errno));
-		return 1;
+	    	// execvp will only return on an error so make sure that we check the errno
+		// and exit with the correct return status for the error that we encountered
+		int status = EXIT_FAILURE;
+		switch errno {
+		case ENOENT:
+			status = 127;
+			break;
+		case EACCES:
+			status = 126;
+			break;
+		}
+		fprintf(stderr, "%s\n", strerror(errno));
```

Comment:
Also the error message is not as important as the return status so take that into consideration as we have some flexibility ;)